package/MDAnalysis/coordinates/chain.py

```diff
@@ -347,14 +513,12 @@ def _read_frame(self, frame):
     def _chained_iterator(self):
         """Iterator that presents itself as a chained trajectory."""
         self._rewind()  # must rewind all readers
-        readers = itertools.chain(*self.readers)
-        for frame, ts in enumerate(readers):
-            ts.frame = frame  # fake continuous frames, 0-based
-            self.ts = ts
-            # make sure that the active reader is in sync
-            i, f = self._get_local_frame(frame)  # uses 0-based frames!
-            self.__activate_reader(i)
-            yield ts
+        for i in range(self.n_frames):
+            j, f = self._get_local_frame(i)
+            self.__activate_reader(j)
+            self.ts = self.active_reader[f]
```

Comment:
This requires the underlying reader to be able to do random access. Do *all* our readers support it nowadays?

Reply:
yes